lib/vagrant-libvirt/templates/domain.xml.erb

```diff
@@ -239,6 +239,13 @@
 	  .join(' ')
 	-%>/>
       </source>
+    <%- if [pci[:guest_domain], pci[:guest_bus], pci[:guest_slot], pci[:guest_function]].any? -%>
+	<address type='pci' <%= pci.select { |k,_| [:guest_domain, :guest_bus, :guest_slot, :guest_function].include? k }
+	  .reject { |k,v| v.nil? }
+	  .map { |k,v| "#{k.to_s[6..-1]}='#{v}'" }
```

Comment:
If at the top of the loop over the pcis you add the line:
```
pci = pci.reject { |k,v| v.nil? }
```
I believe it will remove the need to reject any empty values up front for the entire block. This is a slight tweak to the main pci block, and should allow removing both lines containing `.reject { |k,v| v.nil? }`.

Then by switching the code here to the following should allow skipping the need to explicitly identify the keys.
```suggestion
<%-   prefix = 'guest_'
      pci_guest = pci.select { |k,_| k.to_s.start_with?(prefix) }                                                                            
      if !pci_guest.empty?                                                                                                                   
-%>   
      <address type='pci' <%= pci_guest.map { |k,v| "#{k.to_s[prefix.length..-1]}='#{v}'" }.join(' ') -%>/>
```

If you want to do something similar above where you use `pci.select { |k,_| !k.to_s.start_with?(prefix) }` instead of the explicit keys `[:domain, :bus, :slot, :function]`, that would be great, but not necessary for this PR.

Reply:
Thank you for the detailed response, appreciate it! I've incorporated the change for both the host and guest attributes and it does make it a fair bit cleaner. Will push the changes now, and include the unit-test as well as you suggested, which is passing.